pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -955,7 +956,31 @@ private void resetSubscriptionCursor(Subscription subscription, CompletableFutur
     @Override
     public CompletableFuture<Void> unsubscribe(String subscriptionName) {
         CompletableFuture<Void> unsubscribeFuture = new CompletableFuture<>();
+        getBrokerService().getManagedLedgerFactory().asyncDelete(TopicName.get(MLPendingAckStore
```

Comment:
We should try this only in case we have transactions enabled?
Otherwise we are going to slow down every 'unsubscribe' operation with additional operations (look-up, metadata...)

Reply:
if we enabled after disable, we will not have method to delete this, so we should every time delete `pendingAck`. I think we don't need to think about delete cursor speed.

Reply:
my main concern is about Readers
the reader will unsubscribe the temporary subscription.

but probably the Reader problem is different...should we create pending acks topics for Readers ? and for Non Durable subscriptions ?

Reply:
now, may don't think about the NonPersistentSubscription, this can be considered later.